test/org/sagebionetworks/bridge/services/ConsentServiceImplTest.java

```diff
@@ -104,54 +119,37 @@ public void after() {
+        assertTrue(histories.isEmpty());
         
-        assertTrue(consentService.hasUserConsentedToResearch(testUser.getStudy(), testUser.getUser()));
-        ConsentSignature returnedSig = consentService.getConsentSignature(testUser.getStudy(), testUser.getUser());
-        assertEquals("John Smith", returnedSig.getName());
-        assertEquals("1990-11-11", returnedSig.getBirthdate());
-        assertNull(returnedSig.getImageData());
-        assertNull(returnedSig.getImageMimeType());
-        assertEquals(SharingScope.ALL_QUALIFIED_RESEARCHERS, optionsService.getEnum(testUser.getUser().getHealthCode(), SHARING_SCOPE, SharingScope.class));
-
-        // Withdraw consent and verify.
-        consentService.withdrawConsent(testUser.getStudy(), testUser.getUser(), WITHDRAWAL, UNIX_TIMESTAMP);
-        assertFalse(consentService.hasUserConsentedToResearch(testUser.getStudy(), testUser.getUser()));
         try {
-            consentService.getConsentSignature(testUser.getStudy(), testUser.getUser());
+            consentService.getConsentSignature(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser());
             fail("expected exception");
-        } catch (EntityNotFoundException ex) {
+        } catch (EntityNotFoundException e) {
         }
     }
     
     @Test
-    public void canConsentWithSignatureImage() {
-        // Consent and verify.
+    public void canGiveAndWithdrawConsent() {
+        // Consent and verify, using signature image as well as other information. This combines information 
+        // with another give/withdraw consent test that was redundant.
         ConsentSignature signature = new ConsentSignature.Builder().withName("Eggplant McTester")
                 .withBirthdate("1970-01-01").withImageData(TestConstants.DUMMY_IMAGE_DATA)
                 .withImageMimeType("image/fake").build();
         
         long signedOn = signature.getSignedOn();
         
-        consentService.consentToResearch(testUser.getStudy(), testUser.getUser(), signature, SharingScope.NO_SHARING, false);
-        assertTrue(consentService.hasUserConsentedToResearch(testUser.getStudy(), testUser.getUser()));
-
-        ConsentSignature returnedSig = consentService.getConsentSignature(testUser.getStudy(), testUser.getUser());
+        consentService.consentToResearch(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser(), signature, SharingScope.NO_SHARING, false);
         
+        List<ConsentStatus> statuses = consentService.getConsentStatuses(context);
+        assertFalse(statuses.isEmpty());
+
+        ConsentSignature returnedSig = consentService.getConsentSignature(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser());
         assertEquals("Eggplant McTester", returnedSig.getName());
         assertEquals("1970-01-01", returnedSig.getBirthdate());
         assertEquals(TestConstants.DUMMY_IMAGE_DATA, returnedSig.getImageData());
```

Comment:
I cleaned up this test but didn't see the participant options test, I'll add now.